app/model/SubscriptionData.scala

```diff
@@ -41,28 +61,29 @@ case class SubscriptionData(personalData: PersonalData, paymentData: PaymentData
 object SubscriptionData {
   def fromIdUser(u: IdUser) = {
     implicit class OptField[A](opt: Option[A]) {
-      def getOrDefault[B](get: A => Option[B], default: B): B =
+      def getOrDefault[B](get: A => Option[B], orElse: A => Option[B], default: B): B =
         (for {
           fieldOpt <- opt
-          fieldValue <- get(fieldOpt)
+          fieldValue <- get(fieldOpt) orElse orElse(fieldOpt)
         } yield fieldValue) getOrElse default
-      def getOrBlank(get: A => Option[String]): String = getOrDefault(get, "")
+      def getOrBlank(get: A => Option[String]): String = getOrDefault(get, _ => None, "")
+      def getOrBlank(get: A => Option[String], orElse: A => Option[String]): String = getOrDefault(get, orElse, "")
     }
 
     val addressData = Address(
-      lineOne = u.privateFields.getOrBlank(_.billingAddress1),
-      lineTwo = u.privateFields.getOrBlank(_.billingAddress2),
-      town = u.privateFields.getOrBlank(_.billingAddress3),
-      postCode = u.privateFields.getOrBlank(_.billingPostcode),
-      countyOrState = u.privateFields.getOrBlank(_.country),
-      country = Country.UK
+      lineOne = u.privateFields.getOrBlank(_.billingAddress1, _.address1),
```

Comment:
Turns out this logic is wrong as it would potentially cause to interleaf fields from billing and delivery address, while we should build  `AddressData` from either one or the other.